geoh5py/ui_json/validation.py

```diff
@@ -72,38 +104,64 @@ def workspace(self, value: Workspace | None):
+        val = {}
+        for k in unique_validators:
+            if k not in all_validators:
+                raise ValueError(f"No validator implemented for argument '{k}'.")
+            else:
+                val[k] = all_validators[k]
+        return val
+
+    @staticmethod
+    def infer_validations(ui_json: dict[str, Any]):
+        validations = {}
+        for key, item in ui_json.items():
+            if not isinstance(item, dict):
+                continue
 
-    @validations.setter
-    def validations(self, val):
-        if isinstance(val, dict):
-            validator_list = list({key for item in val.values() for key in item})
-            for key in validator_list:
-                if key == "property_group_type":
-                    self._validators[key] = PropertyGroupValidator()
-                elif key == "required":
-                    self._validators[key] = RequiredValidator()
-                elif key == "shape":
-                    self._validators[key] = ShapeValidator()
-                elif key == "types":
-                    self._validators[key] = TypeValidator()
-                elif key == "values":
-                    self._validators[key] = ValueValidator()
-                elif key == "association":
-                    self._validators[key] = AssociationValidator()
+            if "isValue" in item:
+                validations[key] = {
+                    "types": [UUID, int, float, Entity, type(None)],
+                    "association": item["parent"],
+                }
+            elif "choiceList" in item:
+                validations[key] = {"types": [str], "values": item["choiceList"]}
+            elif "fileType" in item:
+                validations[key] = {
+                    "types": [str],
+                }
+            elif "meshType" in item:
+                validations[key] = {
+                    "types": [UUID, Entity, type(None)],
+                    "association": "geoh5",
+                }
+            elif "parent" in item:
+                validations[key] = {
+                    "types": [UUID, Entity, type(None)],
+                    "association": item["parent"],
+                }
+                if "dataGroupType" in item:
+                    validations[key]["property_group_type"] = item["dataGroupType"]
+                    validations[key]["types"] = [UUID, PropertyGroup, type(None)]
+            elif "value" in item:
+                if item["value"] is None:
+                    check_type = str
                 else:
-                    raise ValueError(f"No validator implemented for argument '{key}'.")
-        elif val is None:
-            self._validators = None
-        else:
-            raise ValueError(
-                "Input 'validations' must be of of type 'dict' or None. "
-                f"Argument of type {type(val)} provided"
-            )
+                    check_type = cast(Any, type(item["value"]))
```

Comment:
What's the use case for this?

Reply:
I don't know, it's your line of code that I moved over from input file class.

Reply:
ahha ok. Must be a reason...